autogen/agentchat/contrib/capabilities/generate_images.py

```diff
@@ -0,0 +1,156 @@
+import abc
+from typing import Any, Dict, List, Literal, Optional, Tuple, Union
+from diskcache import Cache
+import re
+
+from openai import OpenAI
+from autogen.agentchat.agent import Agent
+from autogen.agentchat.contrib import img_utils
+from .agent_capability import AgentCapability
+from autogen.agentchat.contrib.text_analyzer_agent import TextAnalyzerAgent
+
+from PIL.Image import Image
+from autogen import ConversableAgent
+
+SYSTEM_MESSAGE = "You've been given the special ability to generate images."
+DESCRIPTION_MESSAGE = "This agent has the ability to generate images."
+
+
+class ImageGenerator(abc.ABC):
+    @abc.abstractmethod
+    def generate_image(self, prompt: str) -> Image:
+        raise NotImplementedError()
+
+
+class DalleImageGenerator(ImageGenerator):
+    def __init__(
+        self,
+        llm_config: Dict,
+        resolution: Literal["256x256", "512x512", "1024x1024", "1792x1024", "1024x1792"] = "1024x1024",
+        quality: Literal["standard", "hd"] = "standard",
+        num_images: int = 1,
+    ):
+        config_list = llm_config["config_list"]
+        _validate_dalle_model(config_list[0]["model"])
+        _validate_resolution_format(resolution)
+
+        self._model = config_list[0]["model"]
+        self._resolution = resolution
+        self._quality = quality
+        self._num_images = num_images
+        self._dalle_client = OpenAI(api_key=config_list[0]["api_key"])
+
+    def generate_image(self, prompt: str) -> Image:
+        """Got this from Beibin Li's work in agentchat_dalle_and_gpt4v.ipynb"""
+        # Function implementation...
+        cache = Cache(".cache/")  # Create a cache directory
+        key = (self._model, prompt, self._resolution, self._quality, self._num_images)
+
+        # if key in cache:
```

Comment:
Why is cache removed here?

Another possibility is to have a parameter in the __init__: `use_cache` or `cache_setting`.